industry_benchmarks/utils/tests/test_traj_cleanup.py

```diff
@@ -0,0 +1,61 @@
+import MDAnalysis as mda
+import pytest
+from importlib import resources
+from ..traj_cleanup import extract_data
+import numpy as np
+
+
+@pytest.fixture
+def simulation():
+    with resources.files('utils.tests.data.example_traj') as d:
+        yield d / 'simulation.nc'
+
+
+@pytest.fixture
+def checkpoint():
+    with resources.files('utils.tests.data.example_traj') as d:
+        yield d / 'checkpoint.chk'
+
+
+@pytest.fixture
+def pdb():
+    with resources.files('utils.tests.data.example_traj') as d:
+        yield d / 'hybrid_system.pdb'
+
+
+@pytest.fixture
+def outfile():
+    with resources.files('utils.tests.data.example_traj') as d:
+        yield d / 'data.npz'
+
+
+@pytest.fixture
+def out_traj():
+    with resources.files('utils.tests.data.example_traj') as d:
+        yield d / 'out'
+
+
+def test_extract_data(simulation, checkpoint, pdb, outfile, out_traj):
+
+    extract_data(simulation, checkpoint, pdb, outfile, out_traj)
+    data = np.load(outfile)
+    u_ln = data['u_ln']
+    N_l = data['N_l']
+    replicas_state_indices = data['replicas_state_indices']
+    # u_ln array should have a length of 11 (number of lambda windows)
+    assert len(u_ln) == 11
+    # N_l array should have a length of 11 (number of lambda windows)
+    assert len(N_l) == 11
+    # All values in N_l should be equal since we didn't subsample the data
+    assert list(N_l).count(N_l[0]) == len(N_l)
+    # Check the length of the u_ln matrix
+    assert [len(n) == len(u_ln) * N_l[0] for n in u_ln]
```

Comment:
Is it worth at this point also running mbar on this data? (i.e. decorrelate & then calculate an energy value) - that way we check that this works & gives the right value?

Reply:
As discussed in yesterdays call, the example files are from a 250 ps simulation so it may not make sense to look at the DG value. Please let me know if it would make sense to use files from a full simulation instead.